### PR TITLE
Adding catch prob and rating attack-defense DTS

### PR DESCRIPTION
**Pull request by rlodev67**
## Description

This PR is adding catch prob DTS and rating attack-defense DTS.
## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (would cause existing functionality to change)

## Motivation and Context

Since some forks are sending catch probs and movesets rating, it can be usefull to add it on stock PA

## How Has This Been Tested?

Tested on my map lab, on sloppy fork

## Screenshots (if appropriate):
<a href="https://imgbb.com/">github_PA</a>

## Wiki Update

- [X] This change requires an update to the Wiki.
- [ ] This change does not require an update to the Wiki.

## Checklist
- [X] This change follows the code style of this project.
- [X] This change only changes what is necessary to the feature.

### CODE REVIEW

**Comment by JRCrawford** [comment]
Would be really nice to see this feature.

**Comment by SenorKarlos** [comment]
As an administrator and not a programmer I can't comment on the coding, but this feature worked well in previous PA forks and as well in another bot-style announcer. As a supporter I'd like to see this implemented :)

**Comment by billyjbryant** [comment]
In your screenshot you show PokeBalls, are these custom emojis on your setup?  Also, what scanner is this PR based on?

**Comment by rlodev67** [comment]
@billyjbryant , it's just emojis on discord server.
It's based on a fork of RM.

**Comment by billyjbryant** [comment]
Is the data provided also provided by Stock RM?

File: PokeAlarm/Events/MonEvent.py [hunk 1]
```diff
@@ -83,6 +83,18 @@ def __init__(self, data):
...
+        # Catch Probs
+        self.catch_prob_1 = check_for_none(
+            float, data.get('catch_prob_1'), 'unkw')
+        self.catch_prob_2 = check_for_none(
+            float, data.get('catch_prob_2'), 'unkw')
+        self.catch_prob_3 = check_for_none(
+            float, data.get('catch_prob_3'), 'unkw')
+
+        # Attack Rating
+        self.rating_attack = data.get('rating_attack')
```
Comment (kvangent) [inline PokeAlarm/Events/MonEvent.py]: `Unknown` and `check_for_none` here as well.

File: PokeAlarm/Events/MonEvent.py [hunk 2]
```diff
@@ -83,6 +83,18 @@ def __init__(self, data):
         self.charge_duration = get_move_duration(self.charge_id)
         self.charge_energy = get_move_energy(self.charge_id)
 
+        # Catch Probs
+        self.catch_prob_1 = check_for_none(
+            float, data.get('catch_prob_1'), 'unkw')
```
Comment (kvangent) [inline PokeAlarm/Events/MonEvent.py]: You should use the `Unknown` class for these. See the other examples.

File: PokeAlarm/Events/MonEvent.py [hunk 3]
```diff
@@ -226,6 +238,23 @@ def generate_dts(self, locale, timezone, units):
             'weight': self.weight,
             'size': locale.get_size_name(self.size_id),
 
+            # Attack rating
+            'rating_attack': (
+                self.rating_attack.upper() if self.rating_attack else '-'),
```
Comment (kvangent) [inline PokeAlarm/Events/MonEvent.py]:
```python
'rating_attack': Unknown.is_empty(self.rating_attack, Unknown.TINY),
```
If we are always going to call .upper, we should just send the upper the version? Also we need to consider other locales (although perhaps not an issue, we can get some feedback from @MARTINI1 on this)

File: PokeAlarm/Events/MonEvent.py (hunk 1 above)
Comment (rlodev67) [reply PokeAlarm/Events/MonEvent.py]: done

File: PokeAlarm/Events/MonEvent.py (hunk 2 above)
Comment (rlodev67) [reply PokeAlarm/Events/MonEvent.py]: done

File: PokeAlarm/Events/MonEvent.py (hunk 3 above)
Comment (rlodev67) [reply PokeAlarm/Events/MonEvent.py]: I'm already sending it to upper. I put it for some forks whose didn't do that maybe.
btw, there is no need to add locales cause attack rating is something like A+ // A- // B+ etc

**Comment by rlodev67** [comment]
Thanks for your answer,

I didn't planned to PR it in stock RM. But yes, we would change the name of some fields if you want.
If I'm right, catch probs are send by the encounter request.
movesets rating are extacted from https://pokemongo.gamepress.gg/pokemon/150 where 150 = pokemon id.

**Comment by darkelement1987** [comment]
impossible to get the pokeball icons in telegram alarms right? Might as well just use red/blue/yellow circles.

**Comment by rlodev67** [comment]
I changed it
I don't really know how telegram is working, I'm using discord.

**Comment by kvangent** [comment]
@rlodev67 What does the scanner supply for movesets if no moveset rating is available?

**Comment by rlodev67** [comment]
@kvangent  if movesets rating is unavailabke, it's returning '-'

**Comment by darkelement1987** [comment]
base_catch
great_catch
ultra_catch
atk_grade
def_grade

Sounds great

**Comment by rlodev67** [comment]
Changed it
I changed it for data too, so some changes are needed in forks

**Comment by darkelement1987** [comment]
"I changed it for data too, so some changes are needed in forks" Can this last change be made undone lol?

**Pull request merged**